ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlStatementException.java

```diff
@@ -17,34 +17,112 @@
 
 public class KsqlStatementException extends KsqlException {
 
+  public enum Problem {
+    STATEMENT,
+    REQUEST,
+    OTHER;
+  }
+
   private final String sqlStatement;
-  private final boolean isProblemWithStatement;
+  private final Problem problem;
   private final String rawMessage;
+  private final String unloggedDetails;
+  private final String rawUnloggedDetails;
```

Comment:
I wonder if we should have some comments explaining the uses of `unloggedDetails` and `rawUnloggedDetails`?